From db8a437fd3b354f7f172cf8dc6f072577f7da9d6 Mon Sep 17 00:00:00 2001
From: Wyon Bi <bivvy.bi@rock-chips.com>
Date: Mon, 25 Dec 2017 14:21:05 +0800
Subject: [PATCH] drm/panel: simple: Add support for BOE MV270QUM-N10

This adds support for the BOE Corporation MV270QUM-N10 27"
eDP(HBR2, 5.4Gbps) UHD TFT LCD panel, which can be supported
by the simple panel driver.

Change-Id: Ib7df60a7c9a5cd4755a67bedef35a95bcc16a498
Signed-off-by: Wyon Bi <bivvy.bi@rock-chips.com>
---
 .../display/panel/boe,mv270qum-n10.txt        |  9 +++++++
 drivers/gpu/drm/panel/panel-simple.c          | 27 +++++++++++++++++++
 2 files changed, 36 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/display/panel/boe,mv270qum-n10.txt

diff --git a/Documentation/devicetree/bindings/display/panel/boe,mv270qum-n10.txt b/Documentation/devicetree/bindings/display/panel/boe,mv270qum-n10.txt
new file mode 100644
index 000000000000..a97c0791e230
--- /dev/null
+++ b/Documentation/devicetree/bindings/display/panel/boe,mv270qum-n10.txt
@@ -0,0 +1,9 @@
+Boe Corporation 27" UHD TFT LCD panel
+
+It has been designed to apply eDP(HBR2, 5.4Gbps) interface.
+
+Required properties:
+- compatible: should be "boe,mv270qum-n10"
+
+This binding is compatible with the simple-panel binding, which is specified
+in simple-panel.txt in this directory.
diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index c905fe898f82..ebea165db1e2 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -969,6 +969,30 @@ static const struct panel_desc boe_mv238qum_n20 = {
 	},
 };
 
+static const struct drm_display_mode boe_mv270qum_n10_mode = {
+	.clock = 533000,
+	.hdisplay = 3840,
+	.hsync_start = 3840 + 78,
+	.hsync_end = 3840 + 78 + 28,
+	.htotal = 3840 + 78 + 28 + 54,
+	.vdisplay = 2160,
+	.vsync_start = 2160 + 47,
+	.vsync_end = 2160 + 47 + 8,
+	.vtotal = 2160 + 47 + 8 + 7,
+	.vrefresh = 60,
+	.flags = DRM_MODE_FLAG_NVSYNC | DRM_MODE_FLAG_NHSYNC,
+};
+
+static const struct panel_desc boe_mv270qum_n10 = {
+	.modes = &boe_mv270qum_n10_mode,
+	.num_modes = 1,
+	.bpc = 8,
+	.size = {
+		.width = 597,
+		.height = 336,
+	},
+};
+
 static const struct drm_display_mode boe_nv125fhm_n73_mode = {
 	.clock = 72300,
 	.hdisplay = 1366,
@@ -1769,6 +1793,9 @@ static const struct of_device_id platform_of_match[] = {
 	}, {
 		.compatible = "boe,mv238qum-n20",
 		.data = &boe_mv238qum_n20,
+	}, {
+		.compatible = "boe,mv270qum-n10",
+		.data = &boe_mv270qum_n10,
 	}, {
 		.compatible = "boe,nv125fhm-n73",
 		.data = &boe_nv125fhm_n73,
-- 
2.35.3

